src/lib/server/response_builder.cpp

```diff
@@ -0,0 +1,118 @@
+#include "response_builder.hpp"
+#include "lossy_cast.hpp"
+
+namespace opossum {
+
+template <typename SocketType>
+void ResponseBuilder::build_and_send_row_description(
+    std::shared_ptr<const Table> table,
+    const std::shared_ptr<PostgresProtocolHandler<SocketType>>& postgres_protocol_handler) {
+  // Calculate sum of length of all column names
+  uint32_t column_lengths_total = 0;
+  for (auto& column_name : table->column_names()) {
+    column_lengths_total += column_name.size();
+  }
+
+  postgres_protocol_handler->set_row_description_header(column_lengths_total,
+                                                        static_cast<uint16_t>(table->column_count()));
+
+  for (ColumnID column_id{0u}; column_id < table->column_count(); ++column_id) {
+    uint32_t object_id;
+    int16_t type_width;
+
+    switch (table->column_data_type(column_id)) {
+      case DataType::Int:
+        object_id = 23;
+        type_width = 4;
+        break;
+      case DataType::Long:
+        object_id = 20;
+        type_width = 8;
+        break;
+      case DataType::Float:
+        object_id = 700;
+        type_width = 4;
+        break;
+      case DataType::Double:
+        object_id = 701;
+        type_width = 8;
+        break;
+      case DataType::String:
+        object_id = 25;
+        type_width = -1;
+        break;
+      default:
+        Fail("Bad DataType");
```

Comment:
Don't add a default case so that the compiler complains about missing values.

Reply:
I get your point. This makes the compiler complain about `DataType::Bool` and  `DataType::Null`. The latter is not supported by the PostgreSQL protocol. However, both of them don't seem to be supported as a column's data type in Hyrise at all. How do we want to handle those column types?

Reply:
Bool is gone anyway, for Null, we can continue to fail. It's just that I want the fail cases to be explicitly listed, not just defaulted